taichi/ir/ir_builder.cpp

```diff
@@ -44,6 +44,38 @@ void IRBuilder::set_insertion_point_to_false_branch(IfStmt *if_stmt) {
   set_insertion_point({if_stmt->false_statements.get(), 0});
 }
 
+IRBuilder::LoopGuard::~LoopGuard() {
+  if (location_ >= 0 && location_ < loop_->parent->size() &&
+      loop_->parent->statements[location_].get() == loop_) {
+    // faster than set_insertion_point_to_after()
+    builder_.set_insertion_point({loop_->parent, location_ + 1});
```

Comment:
At a high level, what does this condition mean? Something like "the number of statements before the loop statements stays the same"? I'm wondering in what scenario the number of statements changes - it sounds to me that when using the guard the users are only supposed to modify the statements within the guarded scope.

Reply:
> Something like "the number of statements before the loop statements stays the same"?

Yes, exactly. This condition is to prevent some random things like

```C++
auto *loop = builder.create_range_for(zero, ten);
{
  IRBuilder::LoopGuard _(builder, loop);
  builder.create_add(...);
  builder.set_insertion_point_to(some_random_insertion_point);
  builder.create_add(...);  // if some_random_insertion_point is in the same block but before the loop, it may cause trouble...
}  // here I set the insertion point to a *defined* place -- the point after the loop.

Reply:
OK sounds good! Let's add these comments to the code and this PR is ready to merge!

Reply:
> I set the insertion point to a *defined* place -- the point after the loop.

Could you add this comment to the header ? It describes the API's behavior.